Server/Ebenezer/AISocket.cpp

```diff
@@ -1055,22 +1063,23 @@ void CAISocket::RecvUserHP(char* pBuf)
 void CAISocket::RecvUserExp(char* pBuf)
 {
 	int index = 0;
-	int nid = 0;
-	short sExp = 0;
-	short sLoyalty = 0;
+	int userId = GetShort(pBuf, index);
+	short sExp = GetShort(pBuf, index);
+	short sLoyalty = GetShort(pBuf, index);
 
-	nid = GetShort(pBuf, index);
-	sExp = GetShort(pBuf, index);
-	sLoyalty = GetShort(pBuf, index);
-
-	CUser* pUser = (CUser*) m_pMain->m_Iocport.m_SockArray[nid];
-	if (pUser == nullptr)
+	CUser* pUser = (CUser*) m_pMain->m_Iocport.m_SockArray[userId];
+	if (pUser == nullptr || pUser->m_pUserData == nullptr)
```

Comment:
You don't need to check if m_pUserData is nullptr. It will never be nullptr.
It's initialized on startup and is coupled to the instance.